python/cudf/cudf/core/series.py

```diff
@@ -1000,7 +1000,25 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
         if fname in binary_operations:
             not_reflect = self is inputs[0]
             other = inputs[not_reflect]
-            op = f"__{'' if not_reflect else 'r'}{binary_operations[fname]}__"
+
+            # These operators need to be mapped to their inverses when
+            # performing a reflected operation because no reflected version of
+            # the operators themselves exist.
+            ops_without_reflection = {
+                "gt": "lt",
+                "ge": "le",
+                "lt": "gt",
+                "le": "ge",
+                # ne and eq are symmetric
+                "ne": "ne",
+                "eq": "eq",
+            }
+
+            op = binary_operations[fname]
+            if op in ops_without_reflection and not not_reflect:
```

Comment:
`not_reflect` is a little oddly named for a boolean.  Maybe invert it and call it `reflect`? Or `reflect_operator`? Just trying to avoid the awkward double negative in `not not_reflect`.

Reply:
I know, it bothered me a little bit too. I originally chose it because makes the indexing `other = inputs[not_reflect]` work nicely, but at this point it's used in multiple places where I no longer think that benefit offsets the other pieces. I'll swap it back.

Reply:
Done.